legend-engine-xt-serviceStore-grammar/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/HelperServiceStoreBuilder.java

```diff
@@ -326,14 +325,14 @@ else if (typeReference instanceof ComplexTypeReference)
         return pureTypeReference;
     }
 
-    private static Root_meta_external_store_service_metamodel_SecurityScheme compileSecurityScheme(SecurityScheme securityScheme)
+    private static Root_meta_external_store_service_metamodel_SecuritySchemeRequirement compileSecurityScheme(String securitySchemeId, CompileContext context)
     {
-        List<Function<SecurityScheme, Root_meta_external_store_service_metamodel_SecurityScheme>> processors = ListIterate.flatCollect(IServiceStoreCompilerExtension.getExtensions(), ext -> ext.getExtraSecuritySchemeProcessors());
+        List<Function2<String, CompileContext, Root_meta_external_store_service_metamodel_SecuritySchemeRequirement>> processors = ListIterate.flatCollect(IServiceStoreCompilerExtension.getExtensions(), ext -> ext.getExtraSecuritySchemeProcessors());
 
         return ListIterate
-                .collect(processors, processor -> processor.apply(securityScheme))
+                .collect(processors, processor -> processor.apply(securitySchemeId, context))
                 .select(Objects::nonNull)
                 .getFirstOptional()
-                .orElseThrow(() -> new EngineException("Unsupported SecurityScheme - " + securityScheme.getClass().getSimpleName(), securityScheme.sourceInformation, EngineErrorType.COMPILATION));
+                .orElseThrow(() -> new EngineException("Unsupported SecurityScheme - " + securitySchemeId, null, EngineErrorType.COMPILATION));
```

Comment:
lets have source information here, its really needed so that users know where issue is